src/gmpv_control_box.c

```diff
@@ -419,6 +422,7 @@ static void gmpv_control_box_init(GmpvControlBox *box)
 	box->previous_button = gtk_button_new();
 	box->fullscreen_button = gtk_button_new();
 	box->volume_button = gtk_volume_button_new();
+	box->repeat_button = gtk_button_new();
```

Comment:
```suggestion
	box->repeat_button = gtk_toggle_button_new();
```
Use toggle button here so that the user can see whether or not looping is currently enabled.

Reply:
Shouldn't looping be tri-state, loop-off, loop-current, and loop-playlist?

Reply:
I think that it's enough to have loop-off and loop-playlist for this button. This seems to be what most media players do (and presumably the most common use case).